### ISSUE TITLE
convenient support for different recipient_delimiter

### ISSUE THREAD

**Issue by pesimon** [Mailu/Mailu#164]
Coming from a qmail-based setup, my mail addresses are relying on "-" instead of "+" (postfix default) as delimiter between user-extension@domain. 

To change that I've to modify the container configs a rebuild the imap and smtp container. 

```
$ git diff origin/master
diff --git a/dovecot/conf/dovecot.conf b/dovecot/conf/dovecot.conf
index 0f4b04d..4235521 100644
--- a/dovecot/conf/dovecot.conf
+++ b/dovecot/conf/dovecot.conf
@@ -7,6 +7,7 @@ postmaster_address = {{ POSTMASTER }}@{{ DOMAIN }}
 hostname = {{ HOSTNAME }}
 mail_plugins = $mail_plugins quota
 submission_host = smtp
+recipient_delimiter = -

 service dict {
   unix_listener dict {
@@ -127,7 +128,6 @@ service imap-login {

 protocol lmtp {
   mail_plugins = $mail_plugins sieve
-  recipient_delimiter = +
 }

 service lmtp {
diff --git a/postfix/conf/main.cf b/postfix/conf/main.cf
index 366bee3..d52253e 100644
--- a/postfix/conf/main.cf
+++ b/postfix/conf/main.cf
@@ -133,3 +133,5 @@ milter_default_action = tempfail
 ###############
 # Extra Settings
 ###############
+
+recipient_delimiter = -
```

It would be cool to support this via overrides. 
Currently only the top-level param in dovecot could be overwritten but not that one inside lmtp protocol section.

Not sure how to tackle this in a nice way and not have to much includes statements. Ideas?

**Comment by trueserve** [comment]
It looks like this is disabled by default, which is fine. Setting this as an option in `.env`, with a delimiter specifier (`+` being common amongst other providers, such as Google, so might make sense as the default), shouldn't be too difficult. I'll see if I can figure out how to do this unless kaiyou gets to it first.

(I too come from qmail, It's been an interesting change)

**Comment by kaiyou** [comment]
To be honnest, I am working mostly on refactoring the core before I even try to add more features. But I'd be glad to merge if you find some time to implement this :)

**Comment by Znafon** [comment]
By the way, since postfix 2.11 `recipient_delimiter` is a set that can contains multiple characters. I think it also should be checked to disallow the creation of an address mail using one of the characters in `recipient_delimiters` i.e. if `recipient_delimiters` is set to `+-`, the admin ui should alert and abort when trying to create `contact-sales@mydomain.net` for exemple, since it might not give the expected results.

**Comment by pesimon** [comment]
AFAIK, dovecot+lmtp checks if there is an explicit mailbox for `contact-sales` and will deliver to it, even if `contact` exists and  `recipient_delimiters` is set in postfix+dovecot.

**Comment by Znafon** [comment]
@pesimon I don't know the exacts details of the delivery but if your comment is exact the `contact` might want to use `contact-sales` and not receive the mails, that's why I think there should be a warning or an error.

There is two possibilities:
  - mails for `contact-sales` end in `contact-sales` mailbox and `contact` might loose data so in this case, the creation of `contact-sales` should be forbidden
  - mails for `contact-sales` end in `contact` and the creation of `contact-sales` is useless anyway so could be forbidden to avoid confusion.

**Comment by kaiyou** [comment]
This is being implemented as part of #233
